### ISSUE TITLE
Temporarily suspend one optimizer

### ISSUE THREAD

**Issue by turian** [PyTorchLightning/pytorch-lightning#7198]
## 🚀 Feature

It should be possible to temporarily suspend one optimizer.

### Motivation

When training a GAN, if the generator or discriminator becomes too good, one might want to [suspend](https://twitter.com/theshawwn/status/1260476167053869057) that module's optimizer until the other module catches up.

### Pitch

There should be a simple clean way to disable and re-enable a particular optimizer, so that its training steps are completely skipped until they are re-enabled.

### Alternatives

* Tried returning a simple loss, but got an error because the gradient couldn't backprop.
* Tried disabling that module's gradients, but also got an error.
* Considered zero'ing the gradients manually, but that seems really fiddly and gross.

### Additional context

In general, allowing finer control over how multiple optimizers works would be beneficial in many settings.

**Comment by ananthsub** [comment]
@turian https://pytorch-lightning.readthedocs.io/en/latest/common/optimizers.html#manual-optimization should offer finer control over optimizer steps - does this work for your use case?

**Comment by turian** [comment]
@ananthsub Thanks. It also appears that if I return loss = None it toggles the optimizer: https://github.com/PyTorchLightning/pytorch-lightning/pull/5983
